src/common/packager.ts

```diff
@@ -36,50 +36,55 @@ export class Packager {
-                        });
-                    }).done();
-            }
-        });
+        return this.isRunning()
+            .then(running => {
+                if (!running) {
+                    return this.monkeyPatchOpnForRNPackager()
+                        .then(() => {
+                            let args = ["--port", Packager.PORT];
+                            let childEnvForDebugging = Object.assign({}, process.env, { REACT_DEBUGGER: "echo A debugger is not needed: " });
+
+                            Log.logMessage("Starting Packager", outputChannel);
+                            // The packager will continue running while we debug the application, so we can"t
+                            // wait for this command to finish
+
+                            let spawnOptions = { env: childEnvForDebugging };
 
-        return this.awaitStart().then(() => {
-            if (executedStartPackagerCmd) {
-                Log.logMessage("Packager started.", outputChannel);
-            } else {
-                Log.logMessage("Packager is already running.", outputChannel);
-                if (!outputChannel) {
-                    Log.logMessage("Warning: Debugging is not supported if the React Native Packager is not started within VS Code. If debugging fails, please kill other active React Native packager processes and retry.", outputChannel);
+                            // TODO - PROMISE: We need to consume the result of this spawn
+                            new CommandExecutor(this.projectPath).spawnReactPackager(args, spawnOptions, outputChannel).then((packagerProcess) => {
+                                this.packagerProcess = packagerProcess;
+                                executedStartPackagerCmd = true;
+                            });
+                        });
+                }
+            })
+            .then(() =>
+                this.awaitStart())
+            .then(() => {
+                if (executedStartPackagerCmd) {
+                    Log.logMessage("Packager started.", outputChannel);
+                } else {
+                    Log.logMessage("Packager is already running.", outputChannel);
+                    if (!outputChannel) {
+                        // TODO: This warning is printted incorrectly when the packager was started from the command palette. Fix it.
+                        Log.logWarning("Debugging is not supported if the React Native Packager is not started within VS Code. If debugging fails, please kill other active React Native packager processes and retry.", outputChannel);
+                    }
                 }
-            }
 
-            if (this.sourcesStoragePath) {
-                return this.downloadDebuggerWorker().then(() => {
-                    Log.logMessage("Downloaded debuggerWorker.js (Logic to run the React Native app) from the Packager.");
-                });
-            }
-        });
+                if (this.sourcesStoragePath) {
```

Comment:
Just a heads up - I removed this part of the code in my PR. Whoever merges second will have to handle conflicts.